--- a/boot.cmd.generic
+++ b/boot.cmd.generic
@@ -16,7 +16,7 @@
 			fatload ${devtype} ${devnum}:${distro_bootpart} @@UENV_MMC_LOAD_ADDRESS@@ @@UENV_TEXTFILE@@;
 			echo "Importing environment(@@UENV_TEXTFILE@@) from ${boot_target}..."
 			env import -t @@UENV_MMC_LOAD_ADDRESS@@ $filesize
-			if test -n $uenvcmd; then
+			if test -n "$uenvcmd"; then
 				echo "Running uenvcmd ...";
 				run uenvcmd;
 			fi
